network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractIdentifiableImpl.java

```diff
@@ -111,7 +111,7 @@ public Resource<D> getNullableResource() {
     }
 
     public void setResource(Resource<D> resource) {
-        idBeforeRemoval = resource == null ? this.resource.getId() : null;
+        idBeforeRemoval = resource == null && this.resource != null ? this.resource.getId() : null;
```

Comment:
@SlimaneAmar I have problems in network store server test if I do not check that

Reply:
We'll have to do another PR before this one and ask Slimane for review

Reply:
Ok no, it's difficult to isolate this part in another PR...